src/vm/docs/mod.rs

```diff
@@ -271,6 +271,58 @@ Note: This function is only available starting with Stacks 2.1.",
 "#,
 };
 
+const PRINCIPAL_PARSE_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(principal-parse principal-address)",
+    description:
+        "A principal value is a concatenation of two components: a `(buff 1)` *version byte*,
+indicating the type of account and the type of network that this principal can spend tokens on,
+and a `(buff 20)` *public key hash*, indicating the principal's unique identity.
+`principal-parse` will decompose a principal into its component parts, `{version-byte,hash-bytes}`.
+
+This method returns a `Response` that wraps this pair as a tuple.
+
+If the version byte of `principal-address` matches the network (see `is-standard`), then this method
+returns the pair as its `ok` value.
+
+If the version byte of `principal-address` does not match the network, then this method
+returns the pair as its `err` value.
+
+Note: This function is only available starting with Stacks 2.1.",
+    example: r#"
+(principal-parse 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; Returns (ok (tuple (hash-bytes 0x164247d6f2b425ac5771423ae6c80c754f7172b0) (version 0x1a)))
+(principal-parse 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY) ;; Returns (err (tuple (hash-bytes 0xfa6bf38ed557fe417333710d6033e9419391a320) (version 0x16)))
+"#,
+};
+
+const PRINCIPAL_CONSTRUCT_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(principal-construct version-byte hash-bytes)",
+    description: "A principal value is a concatenation of two things: a `(buff 1)` *version byte*,
+indicating the type of account and the type of network that this principal can spend tokens on,
+and a `(buff 20)` *public key hash*, characterizing the principal's unique identity.
+
+`principal-construct` takes as input such a `(buff 1)` `version-byte` and a `(buff 20)`
+`hash-bytes`, and returns a principal.
+
+This function returns a `Response`. On success, the `ok` value is a `Principal`.
+The `err` value is a value tuple with the form `{err_int:UInt,principal_opt:Option<Principal>}`.
+
+If the single-byte `version-byte` is in the valid range `0x00` to `0x1f`, but is not an appropriate
+version byte for the current network, then the error will be `u0`, and `principal_opt` will contain
+`Some<Principal>`, where the wrapped value is the principal.
+
+If the `version-byte` is a `buff` of length less than 1, if the single-byte `version-byte` is a
+value greater than `0x1f`, or the `hash-bytes` is a `buff` of length less than 20, then `err_int`
+will be `u1` and `principal_opt` will be `None`.
+
+Note: This function is only available starting with Stacks 2.1.",
+    example: r#"
+(principal-construct 0x16 0xfa6bf38ed557fe417333710d6033e9419391a320) ;; Returns (err (tuple (error_int u0) (value (some SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY))))
+(principal-construct 0x1a 0xfa6bf38ed557fe417333710d6033e9419391a320) ;; Returns (ok ST3X6QWWETNBZWGBK6DRGTR1KX50S74D3425Q1TPK)
+"#,
+};
```

Comment:
Can you add some examples of the error outcomes?

Reply:
I added one.. I'm wondering if we really need to dwell that much on the error cases in the docs.

Reply:
This is the go-to source of Clarity function behavior, so we *absolutely* must document the error types.